qiita_db/util.py

```diff
@@ -2311,3 +2317,337 @@ def send_email(to, subject, body):
         raise RuntimeError("Can't send email!")
     finally:
         smtp.close()
+
+
+def resource_allocation_plot(file, cname, sname='all'):
+    """Builds resource allocation plot for given filename and jobs
+
+    Parameters
+    ----------
+    file : str, required
+        Builds plot for the specified file name. Usually provided as tsv.gz
+    cname: str, required
+        Specified job type
+    sname: str, optional
+        Specified job sub type.
+
+    Returns
+    ----------
+    matplotlib.pyplot object
+        Returns a matplotlib object with a plot
+    """
+    # Constants
+    COL_NAME = 'samples * columns'
+
+    df = pd.read_csv(file, sep='\t', dtype={'extra_info': str})
+    df['ElapsedRawTime'] = pd.to_timedelta(df.ElapsedRawTime)
+    if sname == "all":
+        _df = df[(df.cName == cname)].copy()
+    else:
+        _df = df[(df.cName == cname) & (df.sName == sname)].copy()
+
+    fig, axs = plt.subplots(ncols=2, figsize=(10, 4), sharey=False)
+    _df.dropna(subset=['samples', 'columns'], inplace=True)
+    _df[COL_NAME] = _df.samples * _df['columns']
+    ax = axs[0]
+    # k a b
+    # bmin = _df['MaxRSSRaw'].min()
+    models = [mem_model1, mem_model2, mem_model3, mem_model4]
```

Comment:
models should be a variable on top of the script?

Reply:
Not sure what this comment means. On top of the script like a global variable?

Reply:
Sorry for not being clear, rm this line and put it at the top of util.py (well under the library imports) as:
```
MODELS = [mem_model1, mem_model2, mem_model3, mem_model4]
```
so it's easy to find and modify. 

Thanks.

Reply:
Cannot add MODELS at the top because the functions are defined at the end of the file.